### ISSUE TITLE
Need to rotate the buildable to be constructed, mouse wheel is best candidate for that.

### ISSUE THREAD

**Issue by illwieckz** [Unvanquished/Unvanquished#1185]
It would be cool to be able rotate the buildable to be constructed, the mouse wheel is best candidate for that.

**Comment by slipher** [comment]
That would be cool indeed.

It occurs to me that it would also be cool for mouse wheel (or some other button) to change which buildable you are placing. Anything that requires opening a menu is pretty bad, especially for humans, since you can get severely nibbled by a dretch due to a half-second delay. Although without a menu it might be unobvious to new players that you can build more than one type of building.

**Comment by DolceTriade** [comment]
Tutorials!

**Comment by illwieckz** [comment]
> It occurs to me that it would also be cool for mouse wheel (or some other button) to change which buildable you are placing.

Yes, I thought about this, especially since it would fix #1195 (_Opening build menu disable wallwalk when using non-toggled wallwalk_), but it's better to have an obvious build menu pop-up and mouse-roll buildable rotation: if's not a big issue if player doesn't figure how to rotate buildable, it's a big issue if player doesn't figure how to rotate buildables.

Also at some point the tutorial will take entire screen…

**Comment by slipher** [comment]
Now that I am remembering the fact that buildable bounding boxes are always axis-aligned, maybe it is not so interesting to be able to rotate them... combined with the fact that all of our buildables have a more or less square shape and can be used from any direction, it means that the rotation makes no difference whatsoever. The models do appear rotated, but the bounding box is still axis-aligned.

**Comment by illwieckz** [comment]
Hmm, I did not care about hitbox when I opened that issue. My concern is the rotation of the model itself, for example the direction you face when you spawn. This is good to be able to decide the orientation of the spawn.

Also, even if turrets and rocket pods can rotate, getting them oriented right from the start is not bad. That would be for free.

What I would like to see if we can rotate buildable at build time is to change the firing angle of the trapper to something like 45° so it becomes useful even on a vertical wall above a door. Currently the trapper requires the map to be designed for it, there is two problems for it: 1. players are not really the builders, 2. even in stock maps few of them can make something useful out of trappers.

**Comment by slipher** [comment]
So the pitch angle is the one you want to change?

**Comment by illwieckz** [comment]
When you're building on an horizontal floor, I'm speaking about using the vertical axis that go through the center of the buildable to rotate the buildable. All other angles are already defined by the surface you build on anyway…

I'm not sure that's called the _pitch_, you made me doubt, but when I translate « roulis, tangage, lacet » with google translate I get « roll, pitch, yaw », and I'm speaking about the lacet (which seems to translates to yaw).

**Comment by slipher** [comment]
OK, the yaw angle, i.e. the same one which you could control by moving around the target in a circle, is what I originally thought. But then my original contention applies: 

> all of our buildables have a more or less square shape and can be used from any direction

I double-checked the trapper code to make sure this is the case. I believe this check is the only one in which the buildable orientation matters:
```
if ( DotProduct( distance, self->s.origin2 ) < LOCKBLOB_DOT )
{
		return false;
}
```
But `origin2` is determined by the normal vector of the surface. So turning the buildable does not do anything for trappers.

**Comment by bmorel** [comment]
> opening a menu is pretty bad

I agree, and I think it could be made slower if the menu could only stay shown as long as the key is pressed, so that if you need to quickly react, you can simply release the key and are ready to fight anew. Such feature would also be useful for the beacons, which are currently just too slow to be used imo.
Unvanquished menus even feature empty spaces between elements (including the new ones, which are still a considerable improvements since they are centered and keys can be used).
